tests/test_sequence.py

```diff
@@ -137,6 +138,288 @@ def test_magnetic_field():
+        + " have different Rydberg levels.",
+    ):
+        seq.switch_device(IroiseMVP, True)
+
+    # Different Channels basis
+    test_device1.change_rydberg_level(IroiseMVP.rydberg_level)
+    seq = Sequence(reg, test_device1)
+    seq.declare_channel("digital", "raman_global", "q1")
+    with pytest.raises(
+        TypeError,
+        match="No match for channel digital with the"
+        + " right basis and addressing.",
+    ):
+        seq.switch_device(IroiseMVP)
+
+    # Different addressing channels
+    test_device1.change_rydberg_level(test_device2.rydberg_level)
+    seq = Sequence(reg, test_device1)
+    seq.declare_channel("ising", "raman_global")
+
+    with pytest.raises(
+        TypeError,
+        match="No match for channel ising with the"
+        + " right basis and addressing.",
+    ):
+        seq.switch_device(test_device2)
+
+    # Channel addressing and basis issue
+    test_device2.change_rydberg_level(IroiseMVP.rydberg_level)
+    seq = Sequence(reg, test_device2)
+    seq.declare_channel("ising", "rmn_local")
+
+    with pytest.raises(
+        TypeError,
+        match="No match for channel ising with the"
+        + " right basis and addressing.",
+    ):
+        seq.switch_device(IroiseMVP)
+
+    # Strict: Jump_phase_time & CLock-period criteria
+    # Jump_phase_time check 1: phase not nill
+    seq = Sequence(reg, test_device3)
+    twin_seq = Sequence(reg, test_device1)
+
+    seq.declare_channel("ising", channel_id="rydberg_global")
+    twin_seq.declare_channel("ising", channel_id="rydberg_global")
+    seq.add(rise, "ising")
+    twin_seq.add(rise, "ising")
+    seq.add(sweep, "ising")
+    twin_seq.add(sweep, "ising")
+    new_seq = seq.switch_device(test_device1, True)
+
+    assert sample(new_seq).__repr__() == sample(twin_seq).__repr__()
+
+    with pytest.raises(
+        ValueError,
+        match="No channel match for channel ising"
+        + " with the right phase_jump_time & clock_period.",
+    ):
+        seq.switch_device(MockDevice, True)
+
+    # Jump_phase_time check 2: No phase
+    seq = Sequence(reg, test_device3)
+    seq.declare_channel(name="ising", channel_id="rydberg_global")
+    seq.add(rise, "ising")
+    seq.add(fall, "ising")
+
+    with pytest.warns(
+        UserWarning,
+        match="The phase_jump_time of the matching channel"
+        + "on the the new device is different, take it in account"
+        + " for the upcoming pulses.",
+    ):
+        seq.switch_device(Chadoq2, True)
+
+    # Clock_period not match
+    test_device1.change_rydberg_level(test_device2.rydberg_level)
+    seq = Sequence(reg, test_device1)
+    seq.declare_channel("ising", "rydberg_global")
+
+    with pytest.raises(
+        ValueError,
+        match="No channel match for channel ising"
+        + " with the right phase_jump_time & clock_period.",
+    ):
+        seq.switch_device(test_device2, True)
+
+    test_device2.change_rydberg_level(test_device3.rydberg_level)
```

Comment:
Same issue here, please avoid using `change_rydberg_level()` on the `Device` class since this will be deprecated soon